src/index/writer.rs

```diff
@@ -95,14 +101,53 @@ impl<T: Fields + PartialOrd + PartialEq + Debug> IndexWriter<T> {
         let mut byte_writer = factory.create_byte_writer(stream.clone()).await;
         byte_writer.seek_to_tail().await;
 
+        let segments = factory
+            .controller_client()
+            .get_current_segments(&stream)
+            .await
+            .expect("get current segments");
+        assert_eq!(
+            1,
+            segments.key_segment_map.len(),
+            "Index stream is configured with more than one segment"
+        );
+        let segment_with_range = segments.key_segment_map.iter().next().unwrap().1.clone();
+        let segment_name = segment_with_range.scoped_segment;
+
+        let controller_client = factory.controller_client();
+
+        let endpoint = controller_client
+            .get_endpoint_for_segment(&segment_name)
+            .await
+            .expect("get endpoint for segment");
+
+        let raw_client = factory.create_raw_client_for_endpoint(endpoint);
+        let token = controller_client
+            .get_or_refresh_delegation_token_for(stream.clone())
+            .await
+            .expect("controller error when refreshing token");
+        let segment_name = segment_name.to_string();
+        let request = Requests::UpdateSegmentAttribute(UpdateSegmentAttributeCommand {
+            request_id: get_request_id(),
+            segment_name: segment_name.clone(),
+            attribute_id: INDEX_RECORD_SIZE_ATTRIBUTE_ID,
+            new_value: T::get_record_size() as i64,
+            expected_value: i64::MIN,
+            delegation_token: token,
+        });
+        raw_client.send_request(&request).await.map_or_else(
+            |e| panic!("failed to get reply: {}", e),
+            |r| info!("UpdateSegmentAttribute : {:?}", r),
+        );
```

Comment:
This change does not address my comment. The method send_request returns a Result where the value is the reply message from the server. You need to do a match on the reply from the server to make sure it is a AttributeUpdated.